google/cloud/spanner/internal/partial_result_set_source.h

```diff
@@ -51,6 +51,32 @@ class PartialResultSetSource : public internal::ResultSetSource {
     }
     return {};
   }
+
+  std::int64_t RowsModified() const override {
```

Comment:
(Just a comment FTR, not blocking this PR at all.)

Scott and I chatted offline about this. In a follow up PR, the `ExecuteDml` call should be changed to call the non-streaming `ExecuteSql` gRPC method. This will simplify the implementation of our `ExecuteDmlImpl`, and it will eliminate our need to use `PartialResultSetSource` in the DML cases. 

No change in this PR is needed now that I understand the direction we're headed. But maybe you want to drop a comment here that this function will be removed?

Reply:
created issue #929